rfc-capi-for-kvm-capk/README.md

```diff
@@ -0,0 +1,449 @@
+  provisioning tooling for on-prem management clusters.
+- As a GS employee, I want KVM operators to implement the CAPI provider contract so that we can be in the list of
+  CAPI providers to improve the reputation of our company.
+- As a GS on-premises (on-prem) customer, I want KVM operator to implement CAPI so that I have access to new
+  provider-independent features developed by other GS teams.
+- As a GS oncall engineer, I want on-prem installations to be as similar as possible to the other providers so that I
+  can share best practices, on call routines, and documentation with the other teams.
+- As a GS on-prem customer, I want GS to implement a migration strategy from current kvm clusters to CAPI KVM clusters
+  so that I can access the new features without service disruption and workload migration.
+- As a GS on-prem customer, I want GS to implement a break out/migration strategy for on-prem installations so that I
+  can use other CAPI on-prem providers in the future such as CAPV (Cluster API for VSphere (VMWare)) or metal3 (Cluster
+  API for bare metal).
+- As a member of Team Rocket, I would like to have confidence in my changes to the platform through the use of reliable
+  unit, integration, and end-to-end tests.
+
+
+### Rationale
+
+With the current move to full CAPI adoption in Team Firecracker and Team Celestial, Team Rocket would be the only user
+and therefore the only maintainer of the current components and code that would be legacy for the other teams.
+
+This would mean that we would need to put a lot of effort into understanding and owning this code fully in order to be
+able to fix bugs and maintain KVM installations
+in the future.
+
+Instead of being the sole maintainer of a huge amount of legacy code, Team Rocket wants to invest this time in
+implementing CAPI for KVM operator and maintain relative platform parity with the other GS provider teams.
+
+### Goals
+
+- Customers can upgrade existing clusters to CAPK without significant workload downtime.
+- Existing CAPI-compatible tools such as kubectl-gs, happa, GS api, and clusterctl work with CAPK clusters.
+- Team Rocket is responsible for a significantly reduced number of repos and lines of code.
+
+### Non-goals
+
+- Get CAPK accepted as a CNCF project.
+- Support mixed CAPK/CAPV clusters.
+- Support mixed management clusters (e.g. CAPK on AWS).
+- Support `MachineSet` or `MachinePool`.
+- Support CAPI <=v1alpha3.
+
+### Scope
+
+We outlined all existing features in the GS KVM platform to define the scope of the implementation. Many features of
+platform can be provided by CAPI now. We therefore split the implementation into features that we must migrate to the
+new operator and those we must ensure are handled by CAPI as follows:
+
+#### Features to migrate to new operator
+
+- DNS and NTP configuration for WC nodes (OS level)
+- iSCSI initiator name (control plane nodes)
```

Comment:
What for? etcd?

Reply:
It seems that this is actually passed in to masters and workers to fix a kubernetes bug. Here's our PR for it https://github.com/giantswarm/kvm-operator/pull/643 This is used for a systemd unit so we should be able to handle it in the `KubeadmConfig` so it doesn't need to be part of the operator. I'll drop it for now.